app/src/main/java/it/niedermann/nextcloud/deck/persistence/sync/SyncManager.java

```diff
@@ -166,76 +172,130 @@ public void onError(Throwable throwable) {
 
     @AnyThread
     public void synchronize(@NonNull IResponseCallback<Boolean> responseCallback) {
-        if(responseCallback.getAccount() == null) {
+        synchronize(Collections.singletonList(responseCallback));
+    }
+    private void synchronize(@NonNull List<IResponseCallback<Boolean>> responseCallbacks) {
+        IResponseCallback responseCallback = responseCallbacks.get(0);
+        Account callbackAccount = responseCallback.getAccount();
+        if(callbackAccount == null) {
             throw new IllegalArgumentException(Account.class.getSimpleName() + " object in given " + IResponseCallback.class.getSimpleName() + " must not be null.");
         }
-        if(responseCallback.getAccount().getId() == null) {
+        Long callbackAccountId = callbackAccount.getId();
+        if(callbackAccountId == null) {
             throw new IllegalArgumentException(Account.class.getSimpleName() + " object in given " + IResponseCallback.class.getSimpleName() + " must contain a valid id, but given id was null.");
         }
-        doAsync(() -> refreshCapabilities(new IResponseCallback<Capabilities>(responseCallback.getAccount()) {
-            @Override
-            public void onResponse(Capabilities response) {
-                if (!response.isMaintenanceEnabled()) {
-                    if (response.getDeckVersion().isSupported(appContext)) {
-                        long accountId = responseCallback.getAccount().getId();
-                        Date lastSyncDate = LastSyncUtil.getLastSyncDate(responseCallback.getAccount().getId());
-                        Date now = DateUtil.nowInGMT();
+        List<IResponseCallback<Boolean>> queuedCallbacks = RUNNING_SYNCS.get(callbackAccountId);
+        if (queuedCallbacks != null){
+            queuedCallbacks.addAll(responseCallbacks);
+            return;
+        } else {
+            queuedCallbacks = new ArrayList<>();
+            queuedCallbacks.addAll(responseCallbacks);
```

Comment:
`queuedCallbacks = new ArrayList<>(responseCallbacks);`

Reply:
absolutely. but the second one would be an NPE ;)

Reply:
Yes, both variants would produce `NPE`, in `constructor` as well as in `addAll`. You will need a `null` check. The second way was more meant to be more performant because the `ArrayList` is reused instead of instantiated.